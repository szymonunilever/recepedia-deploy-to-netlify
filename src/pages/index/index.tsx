import { graphql } from 'gatsby';
import React from 'react';
import Layout from 'src/components/Layout/Layout';
import SEO from 'src/components/Seo/Seo';
import { Text, TagName } from 'src/components/lib/components/Text';

import { RecipeListing } from 'src/components/lib/components/RecipeListing';
import Hero from 'src/components/lib/components/Hero';
import { findPageComponentContent } from 'src/utils';
import { RecipeItem } from 'src/components/lib/components/RecipeListing/partials';

const HomePage = ({ data }: HomePageProps) => {
  const page = data.allPage.edges[0].node;
  const components = page.components.items;
  const recipes = data.allRecipe.nodes;

  return (
    <Layout>
      <SEO title="Recepedia Home" />

      <Text tag={TagName['h1']} text={data.allPage.edges[0].node.title} />
      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'LatestAndGreatest'
          )}
          list={recipes}
        />
      </section>

      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'TopRecipes'
          )}
          list={recipes}
        />
      </section>

      <section>
        <Hero
          content={findPageComponentContent(components, 'Hero')}
          viewType="Image"
        />
      </section>
    </Layout>
  );
};

export default HomePage;

export const pageQuery = graphql`
  {
    allPage(filter: { type: { eq: "Home" } }) {
      edges {
        node {
          ...PageFields
        }
      }
    }
    allRecipe(skip: 10) {
      nodes {
        ...RecipeFields
      }
    }
  }
`;

interface HomePageProps {
  data: {
    allPage: {
      edges: Edge<PageNode>[];
    };
    allRecipe: {
      nodes: RecipeItem[];
    };
  };
}

interface Edge<T> {
  node: T;
}

interface PageNode {
  components: {
    items: {
      name: string;
      content: any;
    }[];
  };
  title: string;
  type: string;
}
