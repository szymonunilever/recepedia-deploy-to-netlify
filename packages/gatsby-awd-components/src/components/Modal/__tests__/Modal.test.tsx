import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import { Modal, ModalProps } from '../index';
import toJson from 'enzyme-to-json';

describe('<Modal />', () => {
  let wrapper: ReactWrapper;
  const props: ModalProps = {
    children: <div>Content</div>,
    isOpen: true,
  };

  beforeEach(() => {
    // eslint-disable-next-line react/no-children-prop
    wrapper = mount(<Modal {...props} children={props.children} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
