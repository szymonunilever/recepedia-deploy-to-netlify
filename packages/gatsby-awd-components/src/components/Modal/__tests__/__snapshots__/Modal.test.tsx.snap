// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> matches the snapshot 1`] = `
<Modal
  isOpen={true}
>
  <Modal
    aria={
      Object {
        "describedby": "modal__description",
        "labelledby": "modal__heading",
        "modal": "true",
      }
    }
    ariaHideApp={true}
    bodyOpenClassName="modal--open"
    className="modal__content modal__content"
    closeTimeoutMS={0}
    isOpen={true}
    overlayClassName="modal__overlay modal__overlay"
    parentSelector={[Function]}
    portalClassName="modal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <Portal
      containerInfo={
        <div
          class="modal"
        >
          <div
            class="ReactModal__Overlay ReactModal__Overlay--after-open modal__overlay modal__overlay"
          >
            <div
              aria-describedby="modal__description"
              aria-labelledby="modal__heading"
              aria-modal="true"
              class="ReactModal__Content ReactModal__Content--after-open modal__content modal__content"
              role="dialog"
              tabindex="-1"
            >
              
              <div
                id="modal__description"
              >
                <div>
                  Content
                </div>
              </div>
            </div>
          </div>
        </div>
      }
    >
      <ModalPortal
        aria={
          Object {
            "describedby": "modal__description",
            "labelledby": "modal__heading",
            "modal": "true",
          }
        }
        ariaHideApp={true}
        bodyOpenClassName="modal--open"
        className="modal__content modal__content"
        closeTimeoutMS={0}
        defaultStyles={
          Object {
            "content": Object {
              "WebkitOverflowScrolling": "touch",
              "background": "#fff",
              "border": "1px solid #ccc",
              "borderRadius": "4px",
              "bottom": "40px",
              "left": "40px",
              "outline": "none",
              "overflow": "auto",
              "padding": "20px",
              "position": "absolute",
              "right": "40px",
              "top": "40px",
            },
            "overlay": Object {
              "backgroundColor": "rgba(255, 255, 255, 0.75)",
              "bottom": 0,
              "left": 0,
              "position": "fixed",
              "right": 0,
              "top": 0,
            },
          }
        }
        isOpen={true}
        overlayClassName="modal__overlay modal__overlay"
        parentSelector={[Function]}
        portalClassName="modal"
        role="dialog"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
        style={
          Object {
            "content": Object {},
            "overlay": Object {},
          }
        }
      >
        <div
          className="ReactModal__Overlay ReactModal__Overlay--after-open modal__overlay modal__overlay"
          onClick={[Function]}
          onMouseDown={[Function]}
          style={Object {}}
        >
          <div
            aria-describedby="modal__description"
            aria-labelledby="modal__heading"
            aria-modal="true"
            className="ReactModal__Content ReactModal__Content--after-open modal__content modal__content"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            role="dialog"
            style={Object {}}
            tabIndex="-1"
          >
            <div
              id="modal__description"
            >
              <div>
                Content
              </div>
            </div>
          </div>
        </div>
      </ModalPortal>
    </Portal>
  </Modal>
</Modal>
`;
